cleanlab_cli/dataset/image_utils.py

```diff
@@ -1,16 +1,33 @@
+import os.path
+
 from cleanlab_cli.types import ImageFileExtension
 from PIL import Image
+from pathlib import Path
+
+from cleanlab_cli.util import get_image_file_extension
+
+
+def get_image_filepath(image_filepath: str, dataset_filepath: str) -> str:
+    if os.path.isabs(image_filepath):
+        return image_filepath
+    else:
+        dataset_path = Path(dataset_filepath)
+        directory_path = dataset_path.parent.absolute()
+        return os.path.join(directory_path, image_filepath)
+
+
+def image_file_exists(image_filepath: str, dataset_filepath: str) -> bool:
+    return os.path.exists(get_image_filepath(image_filepath, dataset_filepath))
 
 
-def is_valid_image(filepath: str) -> bool:
+def image_file_readable(image_filepath: str, dataset_filepath: str) -> bool:
     """
-    valid == has extension .jpeg or .png and image file can be opened by Pillow
-    :param filepath:
-    :return:
+    readable == has extension .jpeg or .png and image file can be opened by Pillow
     """
     try:
-        ImageFileExtension(filepath)
-        Image.open(filepath)
+        image_filepath = get_image_filepath(image_filepath, dataset_filepath)
+        get_image_file_extension(image_filepath)
+        Image.open(image_filepath)
```

Comment:
@ryansingman based on how you've implemented image training, is this check for file readability adequate?

Reply:
Yes, that's sufficient.

From the perspective of training containers, we don't care about the format of the images (as long as they're readable by PIL, so the `get_image_file_extension` check wouldn't be needed for training.